### PR TITLE
chore: go to my collection artwork post-submission

### PR DESCRIPTION
**Pull request by MounirDhahri**
**Although this is 95% reviewable, I believe we should not merge it given the image processing issues which make the UX quite bad**

This PR resolves [ONYX-1007] 

### Description

This PR adds go to my collection artwork post-submission logic. 
This wasn't possible in the past because we didn't get my collection artwork id, but we have it now.

https://github.com/artsy/eigen/assets/11945712/99685735-a198-46db-8230-84197e04683a

Unfortunately, the experience is not ideal currently because we have issues with images processing. This is why **I suggest keeping this PR as a draft because it will worse the UX**

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- go to my collection artwork post-submission - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1007]: https://artsyproduct.atlassian.net/browse/ONYX-1007?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/SellWithArtsy/ArtworkForm/Components/SubmitArtworkBottomNavigation.tsx [hunk 1]
```diff
@@ -110,13 +112,35 @@ export const SubmitArtworkBottomNavigation: React.FC<{}> = () => {
...
+            loading={loadingMyCollectionArtwork}
+            onPress={async () => {
+              try {
+                setLoadingMyCollectionArtwork(true)
+                trackTappedViewArtworkInMyCollection(values.submissionId)
+                switchTab("profile")
+                if (values.submissionId) {
+                  // Wait for gemini to finish processing for 2 seconds
+                  // Which is under Gemini P99 processing times in prod
+                  await setTimeout(() => {}, 2000)
```
Comment (MounirDhahri) [inline src/app/Scenes/SellWithArtsy/ArtworkForm/Components/SubmitArtworkBottomNavigation.tsx:124]: Screenshot 2024-06-10 at 15 45 24

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (go to my collection artwork post-submission - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against afc7a4ffffb2affaace7a80e527624f052ca77d0
</p>

File: src/app/Scenes/SellWithArtsy/ArtworkForm/Components/SubmitArtworkBottomNavigation.tsx [hunk 2]
```diff
@@ -110,13 +112,35 @@ export const SubmitArtworkBottomNavigation: React.FC<{}> = () => {
...
+                      if (myCollectionArtworkID) {
+                        navigate(`/my-collection/artwork/${myCollectionArtworkID}`)
+                      } else {
+                        popToRoot()
+                      }
+                    })
+                  }
+                }
+              } catch (error) {
+                console.error("Error fetching submission information", error)
```
Comment (olerichter00) [inline src/app/Scenes/SellWithArtsy/ArtworkForm/Components/SubmitArtworkBottomNavigation.tsx:140]: Should we, in the case loading the information fails, navigate the user to the My Collection artwork grid?

**Review by olerichter00 — COMMENTED** [review]
If we want to ensure that the images are not just blank if they are not ready to be loaded, we could add something to indicate that the images are still loading. This way, it would be clear why they don't appear.

**Comment by MounirDhahri** [comment]
@olerichter00 I agree- I asked in #dev-help if there is a way to create a new Sidekiq queue for these photos and maybe then we won't have to deal with this. I am waiting for an answer then I will adjust my approach here

**Pull request closed without merging**